data.base.api/src/main/java/ru/tinkoff/qa/neptune/data/base/api/data/operations/DataOperation.java

```diff
@@ -0,0 +1,345 @@
+package ru.tinkoff.qa.neptune.data.base.api.data.operations;
+
+import org.apache.commons.lang3.StringUtils;
+import org.datanucleus.ExecutionContextImpl;
+import org.datanucleus.api.jdo.JDOPersistenceManager;
+import ru.tinkoff.qa.neptune.core.api.event.firing.annotation.MakeFileCapturesOnFinishing;
+import ru.tinkoff.qa.neptune.core.api.event.firing.annotation.MakeStringCapturesOnFinishing;
+import ru.tinkoff.qa.neptune.core.api.steps.SequentialGetStepSupplier;
+import ru.tinkoff.qa.neptune.data.base.api.DataBaseStepContext;
+import ru.tinkoff.qa.neptune.data.base.api.ListOfDataBaseObjects;
+import ru.tinkoff.qa.neptune.data.base.api.PersistableObject;
+import ru.tinkoff.qa.neptune.data.base.api.queries.SelectASingle;
+import ru.tinkoff.qa.neptune.data.base.api.queries.SelectList;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.Arrays.stream;
+import static java.util.Objects.nonNull;
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toList;
+import static ru.tinkoff.qa.neptune.core.api.steps.StoryWriter.action;
+import static ru.tinkoff.qa.neptune.data.base.api.ConnectionToUse.ConnectionDataReader.getConnection;
+
+/**
+ * This class is designed to perform available operations on stored data such as the inserting/updating/deleting
+ * and return results.
+ *
+ * @param <T> is a type of {@link PersistableObject} to be operated (e.g. inserted, updated or deleted)
+ */
+@SuppressWarnings("unchecked")
+@MakeFileCapturesOnFinishing
+@MakeStringCapturesOnFinishing
+public final class DataOperation<T extends PersistableObject>  extends SequentialGetStepSupplier
+        .GetIterableChainedStepSupplier<DataBaseStepContext, List<T>, Map<JDOPersistenceManager, List<T>>, T, DataOperation<T>> {
+
+    private DataOperation(String description, Function<Map<JDOPersistenceManager, List<T>>, List<T>> originalFunction) {
+        super(description, originalFunction);
+    }
+
+    /**
+     * Updating a single stored record. The record to be updated is selected by query and then updated.
+     *
+     * @param howToSelect is a description of query how to select the record
+     * @param set are instances of {@link UpdateExpression} that describe how to update the record
+     * @param <T> is a type of {@link PersistableObject} to be updated
+     * @return an instance of {@link DataOperation}
+     */
+    public static <T extends PersistableObject> DataOperation<T> updated(SelectASingle<T, ?> howToSelect, UpdateExpression<T>... set) {
+        checkArgument(nonNull(howToSelect), "Please define how to select an object to be updated");
+        checkArgument(nonNull(set), "Please define update-actions");
```

Comment:
Since set.length expression is declared, I think there is no needful to declare this one - checkArgument(nonNull(set), "Please define update-actions");

**If you are agreed - apply it for the others.**

Reply:
Disagree. It is possible to pass a null value as a vararg(array) parameter. 
So when it tries to `set.length` then NPE is thrown.